library/alloc/src/lib.rs

```diff
@@ -62,7 +62,8 @@
     html_root_url = "https://doc.rust-lang.org/nightly/",
     html_playground_url = "https://play.rust-lang.org/",
     issue_tracker_base_url = "https://github.com/rust-lang/rust/issues/",
-    test(no_crate_inject, attr(allow(unused_variables), deny(warnings)))
+    test(no_crate_inject, attr(allow(unused_variables), deny(warnings))),
+    cfg_hide(not(test), not(any(test, bootstrap)), target_has_atomic = "ptr")
```

Comment:
Hmm, why hide `target_has_atomic`? Agreed that the other two are implementation details - should `cfg(bootstrap)` also be hidden?

I'm also slightly worried this balloons for every combination of cfgs - maybe hiding `test` should also hide every predicate of `test` too?

Reply:
Having a hidden cfg also hide all cfg expressions that use that cfg seems likely to be a good approach, I'll try it out.